robot/basestation/static/js/roslibjs/ros-helpers.js

```diff
@@ -188,6 +204,99 @@ function initRosWeb () {
 }
 
 /* functions used in main code */
+
+function getTimestamp () {
+  // next few lines taken and adjusted from roslibjs action server example
+  let currentTime = new Date()
+  let secs = currentTime.getTime()/1000 // seconds before truncating the decimal
+  let secsFloored = Math.floor(secs) // seconds after truncating
+  let nsecs = Math.round(1000000000*(secs-secsFloored)) // nanoseconds since the previous second
+
+  // return a dictionary for the ROS log and a float for the gui console
+  return [{secs : secsFloored, nsecs : nsecs}, currentTime]
+}
+
+function logGeneric (logLevel, message) {
```

Comment:
I would just call it roslog

Reply:
I have two functions instead of one now, one called rosLog and one called publishRosLog. when you have time lemme know if you like how i broke things down or if you prefer things some other way